demos/smart_classroom_demo/cpp_gapi/include/initialize.hpp

```diff
@@ -0,0 +1,378 @@
+    tracker_action_params.max_num_objects_in_track = std::numeric_limits<int>::max();
+    tracker_action_params.objects_type = "action";
+
+    return std::make_tuple(const_params, tracker_reid_params, tracker_action_params);
+}
+
+void printInfo() {
+    slog::info << "InferenceEngine: " << printable(*InferenceEngine::GetInferenceEngineVersion()) << slog::endl;
+    if (!FLAGS_teacher_id.empty() && !FLAGS_top_id.empty()) {
+        slog::err << "Cannot run simultaneously teacher action and top-k students recognition."
+                  << slog::endl;
+    }
+    InferenceEngine::Core ie;
+    std::vector<std::string> devices = {FLAGS_d_act, FLAGS_d_fd, FLAGS_d_lm, FLAGS_d_reid};
+    std::set<std::string> loadedDevices;
+    slog::info << "Device info: " << slog::endl;
+    for (const auto &device : devices) {
+        if (loadedDevices.find(device) != loadedDevices.end())
+            continue;
+        std::cout << printable(ie.GetVersions(device)) << std::endl;
+        loadedDevices.insert(device);
+    }
+}
+
+std::string GetBinPath(const std::string& pathXML) {
+    CV_Assert(pathXML.substr(pathXML.size() - 4, pathXML.size()) == ".xml");
+    std::string pathBIN(pathXML);
+    return pathBIN.replace(pathBIN.size() - 3, 3, "bin");
+}
+
+inline cv::gapi::GNetPackage& operator += (cv::gapi::GNetPackage& lhs, const cv::gapi::GNetPackage& rhs) {
+    lhs.networks.reserve(lhs.networks.size() + rhs.networks.size());
+    lhs.networks.insert(lhs.networks.end(), rhs.networks.begin(), rhs.networks.end());
+    return lhs;
+}
+
+void configNets(const std::string& fd_model_path,
+                const std::string& lm_model_path,
+                const std::string& fr_model_path,
+                const std::string& ad_model_path,
+                      cv::gapi::GNetPackage& networks) {
+    if (!ad_model_path.empty()) {
+       /** Create action detector net's parameters **/
+       std::array<std::string, 7> outputBlobList;
+          outputBlobList = isNetForSixActions(ad_model_path)
+              ? outputBlobList = config::action_detector_6
+              : outputBlobList =  config::action_detector_5;
+       auto action_net = cv::gapi::ie::Params<nets::PersonDetActionRec>{
+           ad_model_path,
+           GetBinPath(ad_model_path),
+           FLAGS_d_act,
+       }.cfgOutputLayers(outputBlobList);
+       networks += cv::gapi::networks(action_net);
+    }
+    if (!fd_model_path.empty()) {
+        /** Create face detector net's parameters **/
+        auto det_net = cv::gapi::ie::Params<nets::FaceDetector>{
+            fd_model_path,
+            GetBinPath(fd_model_path),
+            FLAGS_d_fd,
+        };
+        /** FIXME. OpenVino distribution doesn't yet contain this function. **/
+        /*.cfgInputReshape("data",
+                           {1u, 3u, static_cast<size_t>(FLAGS_inh_fd), static_cast<size_t>(FLAGS_inw_fd)})*/
+        slog::warn << "cfgInputReshape is unsupprted yet in OpenVino distribution" << slog::endl;
```

Comment:
Removed warning but FIXME stayed.

Yes, for Face detection network and demo has two flags for this (inh_fd, inw_fd). CNN is reshaped for this height and width.

Reply:
That means `inh_fd` and `inw_fd` are actually ignored now. Then having the warning wasn't a bad idea, sorry. Could you throw an exception here: it is even better than the warning because simply omitting `inh_fd` and `inw_fd` args fixes that, and failure can't be missed compared to warning.

Reply:
For this demo Assert isn't applicable. SCD haven't flag that regulates this behavior. `inh_fd` and `inw_fd` have default values and aren't zero. I leave warning.

BTW, it is strange. OpenVINO should contain ocv 4.5.2 but it doesn't.(